src/Topologies/topology2d.jl

```diff
@@ -132,3 +132,65 @@ boundary_tag(topology::Topology2D, boundary_name::Symbol) =
     findfirst(==(boundary_name), boundary_names(topology))
 
 boundary_faces(topology::Topology2D, boundary) = topology.boundaries[boundary]
+
+function neighboring_elements(topology::Topology2D, elem)
+
+    # Each elem in a Topology2D will have at most 8 neighboring elements
+    #
+    #    o------o------o------o
+    #    |      |      |      |
+    #    |      |      |      |
+    #    o------o------o------o
+    #    |      |      |      |
+    #    |      |      |      |
+    #    o------o------o------o
+    #    |      |      |      |
+    #    |      |      |      |
+    #    o------o------o------o
+    #
+    #
+    # Except for corner elements of the cube (they will have 7)
+    #
+    #    o------o------o------o
+    #    |      |      |      |
+    #    |      |      |      |
+    #    o------o------o------o
+    #    |      |      |      |
+    #    |      |      |      |
+    #    o------o------o------o
+    #    |      |      |   ^
+    #    |      |      | < these two faces are the same at a cube corner
+    #    o------o------o
+
+    neigh_elems = []
+    op_faces = []
+    # First find the cross elements
+    for f in 1:4
+        (opelem, opface, _) = opposing_face(topology, elem, f)
+        # We look to the right of opface. Hence, subtract -1 from opface
+        opface -= 1
+        if opface == 0
+            opface = 4
+        end
+        push!(op_faces, opface)
+        push!(neigh_elems, opelem)
+    end
+
+    for e in 1:4
+        # If the element is at a boundary, we don't want to check its opposing element and return a 0 elem index
+        if neigh_elems[e] == 0
+            opelem = 0
+        else
+            # Given an element in the cross element, check the opposing on the right/top/left/bottom
+            (opelem, _, _) =
+                opposing_face(topology, neigh_elems[e], op_faces[e])
+            # If the opposing element is already in the list of neighboring elements, we are at a cube corner and we return a 0 elem index
+            if opelem ∈ neigh_elems[1:4]
+                opelem = 0
+            end
+        end
+        push!(neigh_elems, opelem)
+    end
+
+    return Tuple(neigh_elems)
```

Comment:
Returning a `Vector` is probably fine here, especially since the `Tuple` will be type-unstable
```suggestion
    return neigh_elems
```

Reply:
I think in general it would be better to move gradually to a fixed size + seintinal value (ex. -1) for missing values for performance.

Reply:
I switched to a fixed-size Array. I thought that 0 was an ok sentinel value in this case (given that `opposing_face` returns 0 in case of non-existent face). Isn't it?

Reply:
that sounds good, you might want to switch to returning an SVector here (other parts of the mesh api are starting to do that).

Reply:
Thanks. I would leave that for another PR, if we are applying that change consistently

Reply:
The general form is:
```
mvec = MVector(...)
... fill in entries
return SVector(mvec)```